packages/caver-transaction/src/transactionTypes/legacyTransaction/legacyTransaction.js

```diff
@@ -0,0 +1,188 @@
+        this._from = address.toLowerCase()
+    }
+
+    /**
+     * @type {string}
+     */
+    get to() {
+        return this._to
+    }
+
+    set to(address) {
+        if (address !== '0x' && !utils.isAddress(address)) throw new Error(`Invalid address ${address}`)
+        this._to = address.toLowerCase()
+    }
+
+    /**
+     * @type {string}
+     */
+    get value() {
+        return this._value
+    }
+
+    set value(val) {
+        this._value = utils.numberToHex(val)
+    }
+
+    /**
+     * @type {string}
+     */
+    get input() {
+        return this._input
+    }
+
+    set input(input) {
+        if (!input || !utils.isHex(input)) throw new Error(`Invalid input data ${input}`)
+        this._input = utils.toHex(input)
+    }
+
+    /**
+     * @type {string}
+     */
+    get data() {
+        return this._input
+    }
+
+    set data(data) {
+        this.input = data
+    }
+
+    /**
+     * Appends signatures array to transaction.
+     * Legacy transaction cannot have more than one signature, so an error occurs if the transaction already has a signature.
+     *
+     * @override
+     * @param {Array.<string>|Array.<Array.<string>>} sig - An array of signatures to append.
+     */
+    appendSignatures(sig) {
+        if (!utils.isEmptySig(this.signatures))
+            throw new Error(
+                `signatures already defined. ${this.type} cannot include more than one signature. Please use tx.signatures = sigArr to replace.`
+            )
+
+        if (Array.isArray(sig[0])) {
+            if (sig.length > 1) throw new Error(`signatures are too long. ${this.type} cannot include more than one signature.`)
+            sig = sig[0]
+        }
+
+        this.signatures = sig
+    }
+
+    /**
+     * Returns RLP-encoded transaction string(rawTransaction).
+     * @return {string}
+     */
+    getRLPEncodingForTransactionHash() {
+        return RLP.encode([
+            Bytes.fromNat(this.nonce),
+            Bytes.fromNat(this.gasPrice),
+            Bytes.fromNat(this.gas),
+            this.to,
+            Bytes.fromNat(this.value),
+            this.input,
+            this.signatures[0],
+            this.signatures[1],
+            this.signatures[2],
+        ])
+    }
+
+    /**
+     * Returns RLP-encoded transaction string for making signature
+     * @override
+     * @return {string}
+     */
+    getRLPEncodingForSignature() {
+        if (this.gasPrice === undefined)
+            throw new Error(`gasPrice is undefined. Define variable in transaction or use 'transaction.fillTransaction' to fill values.`)
+        if (this.nonce === undefined)
+            throw new Error(`nonce is undefined. Define variable in transaction or use 'transaction.fillTransaction' to fill values.`)
+        if (this.chainId === undefined)
+            throw new Error(`chainId is undefined. Define variable in transaction or use 'transaction.fillTransaction' to fill values.`)
```

Comment:
I think these error checks should be also checked in getRLPEncodingForTransactionHash(). What do you think?

Reply:
Um... i didn't want to check optional values in method defined in each transaction types.
But for correct validation, i think those should be validated in each RLP functions.
I will edit! Thank you !

Reply:
I modified it :)